src/main/java/org/checkerframework/specimin/TargetMethodFinderVisitor.java

```diff
@@ -219,19 +220,44 @@ public Visitable visit(MethodDeclaration method, Void p) {
   @Override
   public Visitable visit(Parameter para, Void p) {
     if (insideTargetMethod) {
-      ResolvedType paraType = para.resolve().getType();
-      if (paraType.isReferenceType()) {
-        String paraTypeFullName =
-            paraType.asReferenceType().getTypeDeclaration().get().getQualifiedName();
-        usedClass.add(paraTypeFullName);
-        for (ResolvedType typeParameterValue : paraType.asReferenceType().typeParametersValues()) {
-          String typeParameterValueName = typeParameterValue.describe();
-          if (typeParameterValueName.contains("<")) {
-            // removing the "<...>" part if there is any.
-            typeParameterValueName =
-                typeParameterValueName.substring(0, typeParameterValueName.indexOf("<"));
+      Type type = para.getType();
+      String paraTypeFullName = "";
+      ResolvedType paramType;
+      if (type.isUnionType()) {
+        for (var param : para.getType().asUnionType().getElements()) {
+          try {
+            paramType = param.resolve();
+            paraTypeFullName =
+                paramType.asReferenceType().getTypeDeclaration().get().getQualifiedName();
+            usedClass.add(paraTypeFullName);
+          } catch (
+              Exception
+                  e) { // This visitor should not have an unsolvable parameter type. Throwing if
+            // exception occurs.
+            throw new Error(e.getMessage());
+          }
+        }
+      } else {
+        if (para.getParentNode().get() instanceof CatchClause) {
```

Comment:
This only works if the immediate parent is a catch clause. A catch clause can contain arbitrary additional nesting blocks - for example, it might contain one or more levels of `if` statements or loops. Please add a test case where a catch block has multiple levels of parents before references one of the exception parameters, and then fix this test so that it handles arbitrary nesting.

Reply:
Added a separate test case with additional nesting blocks. Since only the parameter of the catch is targeted, I thinks it's it safe to use this way. 

```
try {
} catch (ExceptionType e) { // --> para.resolve() fails in this case
}
```

when ExceptionType is used in a method parameter, para.resolve() works normally. I have created an issue on java parser repo for this case.